### PR TITLE
STAT-49 Account Balance API

### PR DESCRIPTION
**Pull request by brianjohnson5972**
#638 
Providing dApp API for retrieving an accounts balance and staking balance (similar to external api's get_account).

### CODE REVIEW

File: contracts/eoslib/account.hpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/**
+* @file token.hpp
+* @copyright defined in eos/LICENSE.txt
+* @brief Defines types and ABI for standard token messages and database tables
```
Comment (heifner) [inline contracts/eoslib/account.hpp]: Update doxygen.

File: contracts/eoslib/account.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#include <eoslib/account.h>
+#include <eoslib/print.hpp>
+
+
+namespace eosio { namespace account {
+  /**
+  *  @defgroup tokens Token API
+  *  @brief Defines the ABI for interfacing with standard-compatible token messages and database tables.
+  *  @ingroup contractdev
+  *
```
Comment (heifner) [inline contracts/eoslib/account.hpp]: Update doxygen

File: contracts/eoslib/account.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    * @ret true if account information is retrieved
+    *
+    * @pre data is a valid pointer to a range of memory at least datalen bytes long
+    * @pre data is a pointer to a balance object
+    * @pre *((uint64_t*)data) stores the primary key
+    *
+    *  Example:
+    *  @code
+    *  balance b;
+    *  b.account = n(myaccount);
```
Comment (heifner) [inline contracts/eoslib/account.h]: N(myaccount)

File: contracts/eoslib/account.hpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  * @{
+  */
+
+/**
+*  @struct eosio::account_balance (must match account_balance defined in wasm_interface.cpp)
+*  @brief The binary structure expected and populated by native balance function.
+*  @ingroup tokens
```
Comment (heifner) [inline contracts/eoslib/account.hpp]: Change ingroup

File: contracts/eoslib/account.hpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  /**
+  * Unstaking balance for this account
+  * @brief Unstaking balance for this account
+  */
+  asset unstaking_balance;
+
+  /**
+  * Time at which last unstaking occurred for this account
+  * @brief Time at which last unstaking occurred for this account
+  */
```
Comment (heifner) [inline contracts/eoslib/account.hpp]: Do we need brief if exactly the same as non-brief? All of these briefs.

File: contracts/test_api/test_account.cpp [hunk 6]
```diff
@@ -0,0 +1,27 @@
+#include <eoslib/account.hpp>
```
Comment (heifner) [inline contracts/test_api/test_account.cpp]: copyright

File: contracts/test_api/test_account.cpp [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+#include <eoslib/eos.hpp>
+#include <eoslib/system.h>
+
+#include "test_api.hpp"
+
+/// eos with 8 digits of precision
+#define EOS_SYMBOL  (int64_t(4) | (uint64_t('E') << 8) | (uint64_t('O') << 16) | (uint64_t('S') << 24))
```
Comment (heifner) [inline contracts/test_api/test_account.cpp]: This is defined in asset.hpp, can't you just include it? BTW the comment is wrong.

File: libraries/chain/wasm_interface.cpp [hunk 8]
```diff
@@ -46,6 +48,42 @@ namespace eosio { namespace chain {
...
+      * @brief Unstaking balance for this account
+      */
+      asset unstaking_balance;
+
+      /**
+      * Time at which last unstaking occurred for this account
+      * @brief Time at which last unstaking occurred for this account
+      */
+      time last_unstaking_time;
+   })
```
Comment (heifner) [inline libraries/chain/wasm_interface.cpp]: No ; ? Maybe the macro adds it.

Comment (brianjohnson5972) [reply libraries/chain/wasm_interface.cpp]: #define PACKED_STRUCT(definition) definition __attribute__((packed));

**Pull request merged**